### DISCUSSION TITLE
Is there a way to generate the same number of topics from the same code in a re-run?

### DISCUSSION THREAD

**Discussion by chonchonj23** [MaartenGr/BERTopic#1468] [Q&A]
Dear Maarten,

Every time I re-run the fit_transform step, I would get a slightly different number of topics from the model.

topics, probs = model.fit_transform(docs)

Is there a way or seed value so that I can get the same number of topics from the model in a re-run?

Thank you very much.

Best regards,
Chon

**Comment by MaartenGr** [comment]
You can by setting the `random_state` in UMAP. There is more information about that in the [FAQ](https://maartengr.github.io/BERTopic/faq.html#why-are-the-results-not-consistent-between-runs).
